plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -339,34 +398,63 @@ private void updateDhcpOptionsForExistingVms(Network network, HostVO nuageVspHos
         }
     }
 
+    private void checkMultipleSubnetsCombinedWithUseData(Network network){
+        if (_ntwkOfferingSrvcDao.listServicesForNetworkOffering(network.getNetworkOfferingId()).contains(Network.Service.UserData.getName()) && _vlanDao.listVlansByNetworkId(network.getId()).size() > 1) {
```

Comment:
Please consider using `isEmpty` rather than a size check to determine whether or not a list has elements.  It is more idiomatic/expressive and less error-prone.